scripts/deploy.ts

```diff
@@ -24,6 +24,7 @@ class DeploymentExecutor {
   dai_token;
   gate;
   erc1155NonTransferable;
+  safeERC20WithPermit;
 
```

Comment:
Do we actually need to deploy the library? Since it has no external methods, a think compiler just takes them and there is no nee for separate library instance.
Even looking further down the code (and in tests), I don't see that you would need to link BosonRouter and this library (as is needed when library has external methods). Or I am missing something?

Reply:
We need to deploy the library so that it can give us the deployed address.
We will use that deployed address to verify the library source code using the `verify.ts` in etherscan.

Reply:
Yes, but why? If we deploy it and verify it, but our boson router is not actually linked to it, what's the purpose then?

I'm pretty sure that the library will be verified as part of the boson router verification and not as standalone.
Similar as for uniswapV3
https://etherscan.io/address/0xe592427a0aece92de3edee1f18e0157c05861564#code

Reply:
I was going by this: https://docs.soliditylang.org/en/v0.7.6/using-the-compiler.html?highlight=library%20linking#library-linking

And also this: https://docs.soliditylang.org/en/v0.7.6/contracts.html?highlight=library#libraries
specifically: "Libraries are similar to contracts, but their purpose is that they are deployed only once at a specific address". Since we have created the library ourselves and are not pointing to and already deployed OpenZeppelin library, it would need to be deployed at some point.
But since the functions are` internal`, you might be right.

Reply:
ok, will remove updates from `deploy.ts` and `verify.ts` in the next commit.